chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb

```diff
@@ -71,6 +71,13 @@
 
   include_recipe "crowbar::prepare-upgrade-scripts"
 
+  bash "set pre-upgrade node attribute"
+    code <<-EOF
+      crm node attribute $(hostname) set pre-upgrade true
+    EOF
+    only_if { node["run_list_map"].key? "pacemaker-cluster-member" }
```

Comment:
Is there a better way of doing this than checking `run_list_map`?  Ideally there should be a helper to do it. In the long term we might end up dynamically varying the run list (even though the new role recipes have gone in the opposite direction), so I'd expect `node["pacemaker"]["elements"]` to be a slightly more stable place to check.  But I'd be interested in @vuntz's opinion here.